tests(gatsby-remark-images): Add tests for backgroundColor (#16623)
gatsbyjs/gatsby: packages/gatsby-remark-images/README.md

| `wrapperStyle`         |         | Add custom styles to the div wrapping the responsive images. Use the syntax for the style attribute e.g. `margin-bottom:10px; background: red;` or a function returning a style string which receives the information about the image you can use to dynamically set styles based on the aspectRatio for example.                                                                                                                                                                                                                                                                                                                  |

> | `backgroundColor`      | `white` | Set the box-shadow color of the image to match the background image of your design.<br /><br />**Note:** To completely disable box-shadow, use `backgroundColor: none`. You can also use `backgroundColor: transparent` for a transparent box-shadow (i.e., `box-shadow: inset 0px 0px 0px 400px transparent`).                                                                                                                                                                                                                                                                                                                    |

Reviewer: There are 2 separate things here:
 - resulting effect (which is background after all)
 - implementation (which uses box-shadow)

Just specifying one of those will always confuse someone, so maybe adjusting this to somehow use `background` to describe effect and `box-shadow` to describe implementation will make it so users looking for `box-shadow` will still find this and people just looking at prop list will not be confused?

Author: hi, my main reason for mentioning `backgroundColor: 'transparent'` and `backgroundColor: 'none'` was because I wanted to remove `box-shadow` entirely from `<img class="gatsby-resp-image-image" />`. I was a bit confused with the `backgroundColor` option because I thought I can only change the box-shadow color, but not remove it. After I changed from `backgroundColor: 'transparent'` to `backgroundColor: 'none'`, I noticed that this would remove the `box-shadow` from the `<img />` element. I thought mentioning this in the doc might help someone else, but I don't want this to confuse anyone. Maybe we should just close this PR. I can work on something else instead.

Author: Actually, maybe we can remove the changes to README, but still keep the unit tests

> | `backgroundColor`      | `white` | Set the box-shadow color of the image to match the background image of your design.<br /><br />**Note:** To completely disable box-shadow, use `backgroundColor: none`. You can also use `backgroundColor: transparent` for a transparent box-shadow (i.e., `box-shadow: inset 0px 0px 0px 400px transparent`).                                                                                                                                                                                                                                                                                                                    |

Reviewer 2: @kimbaudi I think this is useful info and we should include it 👍 However, I agree with @pieh that people don't really need to know implementation details. I'd propose something like this:

```suggestion
| `backgroundColor`      | `white` | Set the background color of the image to match your design.<br /><br />**Note:** To completely remove the background color, set this option to `none`.                                                                                                                                                                                                                                                                                                                    |
```

This keeps your update on how to disable the background color, but removes the info about `backgroundColor: transparent` as that's effectively the same as `backgroundColor: none`. It also removes any mention of box-shadow, as that knowledge isn't required to understand this option. 

What do you think?

Author: Hi I made an edit based on you and pieh's suggestion. Thank you both! I will push the changes.
